### PR TITLE
WIP. py-sonata-network-reduction@0.1.3

### PR DESCRIPTION
**Pull request by asanin-epfl**
Depends on #1054

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-dask-mpi/package.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ class PyDaskMpi(PythonPackage):
     version('2.21.0', sha256='76e153fc8c58047d898970b33ede0ab1990bd4e69cc130c6627a96f11b12a1a7')
     version('2.0.0', sha256='774cd2d69e5f7154e1fa133c22498062edd31507ffa2ea19f4ab4d8975c27bc3')
 
-    depends_on('py-dask@2.2:', type=('build', 'run'))
     depends_on('py-mpi4py', type=('build', 'run'))
     depends_on('py-setuptools', type='build')
+    for ver in ('@2.0.0', '@2.21.0'):
+        depends_on('py-dask{}'.format(ver), type=('build', 'run'), when=ver)
```
Comment (GianlucaFicarelli) [inline var/spack/repos/builtin/packages/py-dask-mpi/package.py]: It seems that `dask-mpi` doesn't follow the same versions of `dask`, and it does not have these restrictions in their requirements:
- https://github.com/dask/dask-mpi/blob/2.0.0/environment.yml
- https://github.com/dask/dask-mpi/blob/2.21.0/environment.yml
- https://github.com/dask/dask-mpi/blob/main/environment.yml

Wouldn't work to add an explicit requirement for `py-dask` in `var/spack/repos/builtin/packages/py-sonata-network-reduction/package.py`, that seems consistent with the `setup.py` in `sonata-network-reduction`?
```
    depends_on('py-dask@2.0:2.21', type='run', when='@0.1.2:')
```

Comment (asanin-epfl) [reply var/spack/repos/builtin/packages/py-dask-mpi/package.py]: nice catch, thank you

**Review by matz-e — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-neuron-reduce/package.py [hunk 2]
```diff
@@ -11,9 +11,10 @@ class PyNeuronReduce(PythonPackage):
 
     homepage = "https://github.com/orena1/neuron_reduce"
     url = "https://pypi.io/packages/source/n/neuron_reduce/neuron_reduce-0.0.6.tar.gz"
-    git = "git@github.com:orena1/neuron_reduce.git"
+    git = "git@github.com:BlueBrain/neuron_reduce.git"
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-neuron-reduce/package.py]: repo is not private anymore 
```suggestion
    git = "https://github.com/BlueBrain/neuron_reduce.git"
```

File: var/spack/repos/builtin/packages/py-neuron-reduce/package.py [hunk 3]
```diff
@@ -11,9 +11,10 @@ class PyNeuronReduce(PythonPackage):
...
 
     version('develop', branch='master')
+    version('0.0.9', commit='b8738a17f0234ddc6470108c1f84fc8b27c4cc0f')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-neuron-reduce/package.py]: nitpick : as we have our own repo now (originally it was under Oren's account only), we should start using tags for new versions.

**Review by pramodk — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-sonata-network-reduction/package.py [hunk 4]
```diff
@@ -24,21 +25,28 @@ class PySonataNetworkReduction(PythonPackage):
...
+    depends_on('py-lxml@4.3.4:4.99', type='run', when='@:0.1.2')
+    depends_on('py-pyyaml@5.3:5.99', type='run')
+    depends_on('py-tqdm@4.34:4.99', type='run')
+    depends_on('py-click@6.7:7.99', type='run')
+    depends_on('py-joblib@0.14:0.99', type='run', when='@:0.1.1')
     depends_on('py-bluepyopt@1.8.68:', type='run')
-    depends_on('py-bglibpy@4.3:', type='run')
-    depends_on('py-morphio@2.3.9:', type='run')
+    depends_on('py-bglibpy@4.3:4.999', type='run')
+    depends_on('py-morphio@2.3.9:2.999', type='run')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-sonata-network-reduction/package.py:36]: Is this already a known constraint?

Comment (asanin-epfl) [reply var/spack/repos/builtin/packages/py-sonata-network-reduction/package.py:36]: Now I restrict all dependencies by next major version for safety. Here is the corresponding [setup.py](https://bbpcode.epfl.ch/browse/code/nse/sonata-network-reduction/tree/setup.py)

**Pull request closed without merging**